dsgrid/utils/spark.py

```diff
@@ -118,8 +118,9 @@ def _read_natively(filename):
 
 
 def _post_process_dataframe(df, cache=False, require_unique=None):
-    if cache:
-        df = df.cache()
+    # TODO This is causing Spark warning messages. Disable until we know why.
+    # if cache:
+    #     df = df.cache()
```

Comment:
Okay--this explains why you removed .cache() above (and I googled and caught up a little on spark dataframe.cache() and .persist()).